src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -151,6 +152,58 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 }
 
 
+static Plan *
+BuildDistinctPlan(Query *masterQuery, Plan *topLevelPlan)
+{
+	bool distinctClausesHashable = true;
+	List *distinctClauseList = masterQuery->distinctClause;
+	Plan *plan = topLevelPlan;
+	List *aggregateTargetList = copyObject(masterQuery->targetList);
+	List *aggregateColumnList = pull_var_clause_default((Node *) aggregateTargetList);
+	ListCell *columnCell = NULL;
+
+	Assert(masterQuery->distinctClause);
+	Assert(!masterQuery->hasDistinctOn);
+
+	distinctClausesHashable = grouping_is_hashable(distinctClauseList);
+
+	foreach(columnCell, aggregateColumnList)
+	{
+		Var *column = (Var *) lfirst(columnCell);
+		column->varno = OUTER_VAR;
+	}
+
+	/*
+	 * Create group by plan with HashAggregate if all distinct
+	 * members are hashable, and query does not have distinct on
+	 * and query does not already have sort plan on distinct clauses.
+	 */
+	if (distinctClausesHashable)
```

Comment:
In `create_distinct_paths`, PostgreSQL first tries `sort-based implementations of DISTINCT`, if not possible tries `hash-based implementations of DISTINCT`. Do we have a specific reason to try hash-based one first?

Reply:
when I looked at plans created by postgres, they always used HashAgg when distinct is used. Therefore I prioritized HashAgg over Unique/Sort.